futures-util/src/future_result/mod.rs

```diff
@@ -0,0 +1,455 @@
+//! Futures
+//!
+//! This module contains a number of functions for working with `Future`s,
+//! including the `FutureExt` trait which adds methods to `Future` types.
+
+use core::mem::Pin;
+
+use futures_core::{task, Future, Poll};
+
+// combinators
+mod and_then;
+mod map_ok;
+mod map_err;
+mod err_into;
+mod or_else;
+mod recover;
+
+/* TODO
+mod join;
+mod select;
+pub use self::join::{Join, Join3, Join4, Join5};
+pub use self::select::Select;
+
+if_std! {
+mod join_all;
+mod select_all;
+mod select_ok;
+pub use self::join_all::{join_all, JoinAll};
+pub use self::select_all::{SelectAll, SelectAllNext, select_all};
+pub use self::select_ok::{SelectOk, select_ok};
+}
+*/
+
+pub use self::and_then::AndThen;
+pub use self::map_ok::MapOk;
+pub use self::map_err::MapErr;
+pub use self::err_into::ErrInto;
+pub use self::or_else::OrElse;
+pub use self::recover::Recover;
+
+impl<F, T, E> FutureResult for F
+    where F: Future<Output = Result<T, E>>
+{
+    type Item = T;
+    type Error = E;
+
+    fn poll_result(self: Pin<Self>, cx: &mut task::Context) -> Poll<F::Output> {
+        self.poll(cx)
+    }
+}
+
+/// A convenience for futures that return `Result` values that includes
+/// a variety of adapters tailored to such futures.
+pub trait FutureResult {
```

Comment:
I don't see the `into_future` method that was brought up in the RFC thread-- are you still planning to add it? Also, I think that this should probably live in futures-core and have a `FutureResultExt` here, since folks will use `FutureResult` as a bound.